cli/src/deno.rs

```diff
@@ -81,9 +83,45 @@ pub async fn run(extension_state: ExtensionState, entry_point: &str) -> Result<(
 }
 
 /// See https://github.com/denoland/deno/blob/main/core/examples/ts_module_loader.rs.
-struct TypescriptModuleLoader;
+struct ExtensionsModuleLoader;
+
+impl ExtensionsModuleLoader {
+    async fn load_from_filesystem(path: &Url) -> Result<String> {
+        let path = path
+            .to_file_path()
+            .map_err(|_| anyhow!("{path:?}: is not a path"))?;
+
+        let extensions_path = extensions_path()?;
+        if !path.starts_with(&extensions_path) {
+            return Err(anyhow!(
+                "`{}`: importing from paths outside of the extension's directory is not allowed",
+                path.to_string_lossy(),
+            ));
+        }
+
+        if path.is_symlink() {
+            return Err(anyhow!(
+                "`{}`: importing from symlinks is not allowed",
+                path.to_string_lossy(),
+            ));
+        }
```

Comment:
Have you actually tested this? I'd be curious if deno resolves symlinks internally or not.

Reply:
I can test it manually; it would be nice to have this be automated, but I'm not sure how we can do that cleanly -- I don't like the idea of having a symlink commited in the repo, if that's something which is at all possible.